test/connectivity.jl

```diff
@@ -10,6 +10,12 @@ add_edge!(g,10,9)
 @test is_connected(HouseGraph())
 
 cc = connected_components(g)
+visitor = LightGraphs.TreeBFSVisitorVector(zeros(Int, nv(g)))
+label = zeros(Int, nv(g))
+label = LightGraphs.connected_components!(label, g)
```

Comment:
Oh, I didn't see the exclamation point. Got it. If it's called by `connected_components` then we don't need to. What do you prefer for an API?